packages/pds/tests/account-deletion.test.ts

```diff
@@ -0,0 +1,228 @@
+import { once, EventEmitter } from 'events'
+import Mail from 'nodemailer/lib/mailer'
+import AtpApi, { ServiceClient as AtpServiceClient } from '@atproto/api'
+import { SeedClient } from './seeds/client'
+import basicSeed from './seeds/basic'
+import { Database } from '../src'
+import * as util from './_util'
+import { ServerMailer } from '../src/mailer'
+import { BlobNotFoundError, BlobStore } from '@atproto/repo'
+import { CID } from 'multiformats/cid'
+
+describe('account deletion', () => {
+  let client: AtpServiceClient
+  let close: util.CloseFn
+  let sc: SeedClient
+
+  let mailer: ServerMailer
+  let db: Database
+  let blobstore: BlobStore
+  const mailCatcher = new EventEmitter()
+  let _origSendMail
+
+  // chose carol because she has blobs
+  let carol
+
+  beforeAll(async () => {
+    const server = await util.runTestServer({
+      dbPostgresSchema: 'account_deletion',
+    })
+    close = server.close
+    mailer = server.ctx.mailer
+    db = server.ctx.db
+    blobstore = server.ctx.blobstore
+    client = AtpApi.service(server.url)
+    sc = new SeedClient(client)
+    await basicSeed(sc, server.ctx.messageQueue)
+    carol = sc.accounts[sc.dids.carol]
+
+    // Catch emails for use in tests
+    _origSendMail = mailer.transporter.sendMail
+    mailer.transporter.sendMail = async (opts) => {
+      const result = await _origSendMail.call(mailer.transporter, opts)
+      mailCatcher.emit('mail', opts)
+      return result
+    }
+  })
+
+  afterAll(async () => {
+    mailer.transporter.sendMail = _origSendMail
+    if (close) {
+      await close()
+    }
+  })
+
+  const getMailFrom = async (promise): Promise<Mail.Options> => {
+    const result = await Promise.all([once(mailCatcher, 'mail'), promise])
+    return result[0][0]
+  }
+
+  const getTokenFromMail = (mail: Mail.Options) =>
+    mail.html?.toString().match(/>(\d{6})</)?.[1]
+
+  let token
+
+  it('requests account deletion', async () => {
+    const mail = await getMailFrom(
+      client.com.atproto.account.requestDelete(undefined, {
+        headers: sc.getHeaders(carol.did),
+      }),
+    )
+
+    expect(mail.to).toEqual(carol.email)
+    expect(mail.html).toContain('Delete your Bluesky account')
+
+    token = getTokenFromMail(mail)
+    if (!token) {
+      return expect(token).toBeDefined()
+    }
+  })
+
+  it('fails account deletion with a bad token', async () => {
+    const attempt = client.com.atproto.account.delete({
+      token: '123456',
+      handle: carol.handle,
+      password: carol.password,
+    })
+    await expect(attempt).rejects.toThrow('Token is invalid')
```

Comment:
Oh now it's decided to work again?! 🤣

Reply:
man idk 😅 

I also ended up refactoring that migration test & got rid of those sketch issues